doc/benchexec-in-container.md

```diff
@@ -0,0 +1,233 @@
+
+Start the image with
+```
+  podman run --security-opt unmask=/sys/fs/cgroup --cgroups=split --security-opt unmask=/proc/* --security-opt seccomp=unconfined -it <tag>
+```
+or
+```
+  docker run --privileged --cap-drop=all -it <tag>
+```
+<!--
+In principle, `--security-opt systempaths=unconfined --security-opt seccomp=unconfined`
+should also be sufficient as Docker arguments,
+but then mounting within the container still fails.
+-->
+
+> **IMPORTANT**: The `--privileged` argument, gives the Docker container *full
+> root access* to the host, so make sure to include the `--cap-drop=all` flag,
+> use this command only with trusted images, and configure your Docker
+> container such that everything in it is executed under a different user
+> account, not as root (more details [below](#securing-docker-execution)).
+> BenchExec is not designed to run as root and does not provide any safety
+> guarantees regarding its container under these circumstances.
+
+With this, you should be able to execute, for example, `runexec echo`
+inside the Docker container. (In case you opted for the `.whl` install, you
+need to execute
+`PYTHONPATH=/opt/benchexec.whl python3 -m benchexec.runexecutor <program>`
+instead.)
+
+In case you want to modify this setup, please consider the notes mentioned
+[below](#background-and-technical-details).
+
+### Securing Docker Execution
+
+When running under Docker, the user which executes the commands by default has
+`UID` 0, i.e. has the permissions of `root` on the host system. For example, if
+you mount a folder into the container, the processes started in the container
+have *full write access* to these files, no matter what their permissions are.
+As such, a small mistake in, e.g., an evaluation script that deletes temporary
+files, could easily wreak havoc on your system. Using Podman directly mitigates
+this issue, as the Podman container runs with the permissions of the current
+user, not `root`, and thus cannot mess with system files (in particular, the
+"`root`" user in a Podman container is "mapped" to your current user).
+
+To significantly increase security of your Docker execution, add
+```docker
+# Create non-root user and set it as default
+RUN useradd -ms /bin/bash user
+WORKDIR /home/user
+USER user
+```
+at the end of your Dockerfile. Note: This is still worse than using Podman
+(e.g. privilege escalation within the container leads to obtaining root
+permissions on the host) but drastically reduces the chance for accidental
+problems.
+
+### Non-interactive Setups
+
+In case you intend to only execute one BenchExec process inside the container,
+the setup simplifies notably. In particular, then, it is sufficient to execute
+the BenchExec process as entry point, without requiring the `init.sh` script.
+For example, if BenchExec is installed in a container with tag `my-benchexec`,
+you can simply execute
+`podman run <same flags as above> -t my-benchexec benchexec <arguments>`
```

Comment:
I would like to have this as a ready-to-use command line like in the current docs, so let's copy the flags here and make it a separate code section with three back-ticks.